ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -727,6 +741,31 @@ void setChunkException(RouterException exception) {
     }
 
     /**
+     * Slice this chunk's data to only include the bytes within the operation's specified byte range.
+     * @param blobData the {@link BlobData} for this chunk.
+     * @return A {@link ByteBuffer} that only includes bytes within the operation's specified byte range.
+     */
+    ByteBuffer filterChunkToRange(BlobData blobData) {
+      ByteBuffer buf = blobData.getStream().getByteBuffer();
+      if (!hasRange) {
+        return buf;
+      }
+      GetBlobRange range = options.getRange();
+      long startOffset = range.getStartOffset(totalSize);
+      long endOffset = range.getEndOffset(totalSize);
+      long blobDataSize = blobData.getSize();
+      long chunkStart = (long) chunkSize * (chunkIndex + firstChunkOffset);
```

Comment:
I will change this variable name to `chunkStartOffset` and `chunkEndOffset`.  These were meant to be the offsets of the bytes at the beginning and end of the current chunk, respectively.